src/detect-msg.c

```diff
@@ -125,89 +125,64 @@ static int DetectMsgSetup (DetectEngineCtx *de_ctx, Signature *s, const char *ms
 #ifdef UNITTESTS
 static int DetectMsgParseTest01(void)
 {
-    int result = 0;
-    Signature *sig = NULL;
     const char *teststringparsed = "flow stateless to_server";
     DetectEngineCtx *de_ctx = DetectEngineCtxInit();
-    if (de_ctx == NULL)
-        goto end;
+    FAIL_IF_NULL(de_ctx);
 
     FILE *fd = SCClassConfGenerateValidDummyClassConfigFD01();
     SCClassConfLoadClassficationConfigFile(de_ctx, fd);
 
-    sig = SigInit(de_ctx, "alert tcp any any -> any any (msg:\"flow stateless to_server\"; flow:stateless,to_server; content:\"flowstatelesscheck\"; classtype:bad-unknown; sid: 40000002; rev: 1;)");
-    if(sig == NULL)
-        goto end;
+    Signature *sig = DetectEngineAppendSig(de_ctx,
+            "alert tcp any any -> any any (msg:\"flow stateless to_server\"; "
+            "flow:stateless,to_server; content:\"flowstatelesscheck\"; "
+            "classtype:bad-unknown; sid: 40000002; rev: 1;)");
+    FAIL_IF_NULL(sig);
 
-    if (strcmp(sig->msg, teststringparsed) != 0) {
-        printf("got \"%s\", expected: \"%s\": ", sig->msg, teststringparsed);
-        goto end;
-    }
+    FAIL_IF(strcmp(sig->msg, teststringparsed) != 0);
+
+    DetectEngineCtxFree(de_ctx);
 
-    result = 1;
-end:
-    if (sig != NULL)
-        SigFree(de_ctx, sig);
-    if (de_ctx != NULL)
-        DetectEngineCtxFree(de_ctx);
-    return result;
+    PASS;
```

Comment:
I was working on a similar issue (for detect/iprep) today, and discussing some doubts with a friend, and we got confused regarding the clean up stage of these tests, once we change from the goto end style to the FAIL/PASS macros...

From what we could see, once we use the FAILs, the method will return immediately, skipping any code related to freeing resources. 

I tried to investigate further a bit, but couldn't tell nor find if that is going to happen elsewhere, when we have the macros API. So I am wondering if that is expected behaviour, because these non-freed resources aren't critical; or if this is a change to be implemented later; or maybe I missed something...

Commenting here, because depending on the answers, there will be changes in this one, as well...

Reply:
You are correct. We do not have to worry about freeing resources in case of failed tests. We just need to make sure though that in case of success, we free all the resources.

Reply:
I see! Thank you. Is that because most tests are expected to pass, so we don't have to worry about memory leaks and the likeṣ?